barretenberg/cpp/src/barretenberg/dsl/acir_format/civc_recursion_constraints.cpp

```diff
@@ -0,0 +1,130 @@
+// === AUDIT STATUS ===
+// internal:    { status: not started, auditors: [], date: YYYY-MM-DD }
+// external_1:  { status: not started, auditors: [], date: YYYY-MM-DD }
+// external_2:  { status: not started, auditors: [], date: YYYY-MM-DD }
+// =====================
+#include "barretenberg/dsl/acir_format/mock_verifier_inputs.hpp"
+#include "barretenberg/dsl/acir_format/recursion_constraint.hpp"
+#include "barretenberg/stdlib/client_ivc_verifier/client_ivc_recursive_verifier.hpp"
+#include "barretenberg/stdlib/primitives/bigfield/bigfield.hpp"
+#include "proof_surgeon.hpp"
+
+namespace acir_format {
+
+using namespace bb;
+
+using Builder = bb::UltraCircuitBuilder; // Builder is always Ultra
+using field_ct = stdlib::field_t<Builder>;
+
+template <typename Builder>
+using HonkRecursionConstraintOutput = bb::stdlib::recursion::honk::UltraRecursiveVerifierOutput<Builder>;
+
+using namespace bb;
+
+/**
+ * @brief Creates a dummy vkey and proof object.
+ * @details Populates the key and proof vectors with dummy values in the write_vk case when we don't have a valid
+ * witness. The bulk of the logic is setting up certain values correctly like the circuit size, number of public inputs,
+ * aggregation object, and commitments.
+ *
+ * @param builder
+ * @param proof_size Size of proof with NO public inputs
+ * @param public_inputs_size Total size of public inputs including aggregation object
+ * @param key_fields
+ * @param proof_fields
+ */
+void create_dummy_vkey_and_proof(Builder& builder,
+                                 size_t proof_size,
+                                 size_t public_inputs_size,
+                                 const std::vector<field_ct>& key_fields,
+                                 const std::vector<field_ct>& proof_fields)
+{
+    using ClientIVCRecursiveVerifier = stdlib::recursion::honk::ClientIVCRecursiveVerifier;
+    using IO = stdlib::recursion::honk::HidingKernelIO<Builder>;
+
+    BB_ASSERT_EQ(proof_size, ClientIVCRecursiveVerifier::StdlibProof::PROOF_LENGTH_WITHOUT_PUB_INPUTS());
+
+    size_t num_inner_public_inputs = public_inputs_size - IO::PUBLIC_INPUTS_SIZE;
+    uint32_t pub_inputs_offset = MegaZKFlavor::has_zero_row ? 1 : 0;
+
+    // Generate mock honk vk
+    // Note: log_circuit_size = CONST_PROOF_SIZE_LOG_N
```

Comment:
Note: I believe this will now need to be `MegaFlavor::VIRTUAL_LOG_N` with Sergei's recent changes